sys/net/link_layer/ieee802154/include/ieee802154_config.h

```diff
@@ -0,0 +1,54 @@
+/**
+* @defgroup config_ieee802154 IEEE802.15.4 configurations
```

Comment:
Mh... Isn't there a way to keep them in their original file and just rename them?

Reply:
I think it makes sense to have all the configurable parameter of each module on a separate header file. Makes it easier to find and edit. Also you can just tell what is the exposed configuration of the module without having to go to the documentation if every module implements a similar file structure.

Reply:
You would have the same effect (though admitingly it requires extra knowledge) by having them just named `CONFIG_…` because you can always find them then by using

```sh
git grep "^#define\s\+\<CONFIG_"
```

Reply:
I totally agree with @miri64 that being able to use grep/ag/rg to search for configuration parameters by looking for `#define`s with a common prefix like `CONFIG_` could be a huge productivity boost.

Reply:
NB there are some comments in the usage of something to fetch all config parameters (see [TF status](https://github.com/RIOT-OS/RIOT/issues/9856), expose default configs). Indeed the `CONFIG_` prefix is a good idea. 

We might still need to add other metadata to the options (type, allowed values, help message, etc) that might need some special format/parsing.

Reply:
I like the idea of moving in the direction of putting everything relevant to a module within the module folder. That's a different scope to what this PR is addressing, though, I guess.